scripts/benchmark_stats.py

```diff
@@ -1,77 +1,84 @@
         for file in self.files:
             with open(file) as fp:
                 for row in DictReader(fp, delimiter=","):
-                    percentile_99th = int(row["99%"])
-                    request_count = int(
-                        row.get("Request Count") or row.get("# requests")
+                    percentile_response_time = int(
+                        row[f"{self.PERCENTILE_TO_USE_FOR_ROUTES}%"]
                     )
-
                     if row["Type"] == "GET":
-                        self.weighted_get_requests.append(
-                            percentile_99th * request_count
-                        )
-                        self._total_get_requests += request_count
+                        self._get_requests.append(percentile_response_time)
                     elif row["Type"] == "POST":
-                        self.weighted_post_requests.append(
-                            percentile_99th * request_count
-                        )
-                        self._total_post_requests += request_count
+                        self._post_requests.append(percentile_response_time)
                     elif row["Name"] == "Aggregated":
+                        request_count = row.get("Request Count") or row.get(
+                            "# requests"
+                        )
                         failure_count = row.get("Failure Count") or row.get(
                             "# failures"
                         )
-                        self._total_failures += int(failure_count)
-                        self._total_requests += request_count
+                        self._total_requests = int(request_count)
+                        self._total_failures = int(failure_count)
+
+                        self._percentiles = defaultdict(int)
+                        for percentile in self.PERCENTILES_TO_REPORT:
+                            self._percentiles[percentile] = int(row[f"{percentile}%"])
 
     @property
     def files(self) -> List[str]:
         return self._files
 
     @property
-    def average_weighted_get(self):
-        return sum(self.weighted_get_requests) / self._total_get_requests
+    def percentiles(self) -> Mapping:
+        return self._percentiles
 
     @property
-    def average_weighted_post(self):
-        return sum(self.weighted_post_requests) / self._total_post_requests
+    def average_get(self) -> int:
+        return int(mean(self._get_requests))
 
     @property
-    def average_weighted_total(self):
-        return (
-            sum(self.weighted_get_requests + self.weighted_post_requests)
-            / self._total_requests
-        )
+    def average_post(self) -> int:
+        return int(mean(self._post_requests))
 
     @property
-    def error_percentage(self):
-        return (self._total_failures * 100) / self._total_requests
+    def error_percentage(self) -> int:
+        return (self._total_failures * 100) // self._total_requests
```

Comment:
Double divide?

Reply:
No, it is Python's floor division operator. However, I have removed it as we want the granularity of the error percentages as a floating point as floor division will remove that given the rate of error should be very low.